web/src/lib/services/nostr.ts

```diff
@@ -0,0 +1,52 @@
+import { getEventHash, getPublicKey, signEvent, Kind } from 'nostr-tools';
+import type { SimplePool } from 'nostr-tools/pool';
+import { UserResume, type User } from "$lib/types/user";
+import { relayUrlList, hasExtension, localStorageNostrPreferPMId } from "$lib/nostr/utils";
+
+const EVENT_KIND_RESUME = 66;
+
+async function createEvent(kind: number, content: any, user: User) {
+    let event: any = {
+        kind,
+        content,
+        tags: [],
+        created_at: Math.floor(Date.now() / 1000),
+    }
+
+    if ((!hasExtension() || localStorage.getItem(localStorageNostrPreferPMId) !== null) && user && user.nostr_private_key) { // using PM-generated identity
+        event.pubkey = getPublicKey(user.nostr_private_key);
+        event.id = getEventHash(event);
+        event.sig = signEvent(event, user.nostr_private_key);
+        return event;
+    } else { // using extension identity
+        event.pubkey = await window.nostr.getPublicKey();
+        event.id = getEventHash(event);
+        return await window.nostr.signEvent(event);
+    }
+}
+
+export async function closePool(pool: SimplePool) {
+    await pool.close(relayUrlList);
+}
+
+export function subscribeResumes(pool: SimplePool, receivedCB: (pubkey: string, resume: UserResume, createdAt: number) => void) {
+    let sub = pool.sub(relayUrlList, [{ kinds: [EVENT_KIND_RESUME] }]);
```

Comment:
Back in the day, relay operators were telling that `subs` should have `since` and `limit` so the queries to their databases don't have to do a full-scan of all the data.

That's why we're doing `export let messagesSince: number = 1672837281;  // January 4th 2023` in the Chats.

I don't know if things changed, but maybe it doesn't hurt.